src/core/agent/agent.ts

```diff
@@ -172,68 +177,120 @@ class Agent {
     );
   }
 
-  async start(): Promise<void> {
+  async start(keriUrl: string): Promise<void> {
     if (!Agent.isOnline) {
-      await this.storageSession.open(walletId);
-      this.basicStorageService = new BasicStorage(
-        this.getStorageService<BasicRecord>(this.storageSession)
-      );
-      this.identifierStorage = new IdentifierStorage(
-        this.getStorageService<IdentifierMetadataRecord>(this.storageSession)
-      );
-      this.credentialStorage = new CredentialStorage(
-        this.getStorageService<CredentialMetadataRecord>(this.storageSession)
-      );
-      this.connectionStorage = new ConnectionStorage(
-        this.getStorageService<ConnectionRecord>(this.storageSession)
-      );
-      this.connectionNoteStorage = new ConnectionNoteStorage(
-        this.getStorageService<ConnectionNoteRecord>(this.storageSession)
-      );
-      this.notificationStorage = new NotificationStorage(
-        this.getStorageService<NotificationRecord>(this.storageSession)
-      );
+      await signifyReady();
+      const bran = await this.getBran();
+      this.signifyClient = new SignifyClient(keriUrl, bran, Tier.low);
+      await this.signifyClient.connect();
+      Agent.isOnline = true;
+      this.agentServicesProps.signifyClient = this.signifyClient;
+      this.agentServicesProps.eventService.emit<KeriaStatusChangedEvent>({
+        type: KeriaStatusEventTypes.KeriaStatusChanged,
+        payload: {
+          isOnline: Agent.isOnline,
+        },
+      });
+    }
+  }
 
+  async bootAndConnect(agentUrls: AgentUrls): Promise<void> {
+    if (!Agent.isOnline) {
       await signifyReady();
       const bran = await this.getBran();
-      // @TODO - foconnor: Review of Tier level.
       this.signifyClient = new SignifyClient(
-        ConfigurationService.env.keri.keria.url,
+        agentUrls.url,
         bran,
         Tier.low,
-        ConfigurationService.env.keri.keria.bootUrl
+        agentUrls.bootUrl
       );
-
-      this.agentServicesProps = {
-        signifyClient: this.signifyClient,
-        eventService: new EventService(),
-      };
-
-      this.peerConnectionStorage = new PeerConnectionStorage(
-        this.getStorageService<PeerConnectionMetadataRecord>(
-          this.storageSession
-        )
-      );
-
+      try {
+        await this.signifyClient.boot();
+      } catch (e) {
+        throw new Error(Agent.KERIA_BOOT_FAILED);
```

Comment:
Can we check the error here - if it has already booted, this is OK and we can continue. We should only error if it cannot boot for another reason

Reply:
Can we also `log.error(e)` here and below